payment/payment.go

```diff
@@ -37,19 +38,20 @@ func FromPublicKey(pubkey *btcec.PublicKey, net *network.Network) *Payment {
 		tmpNet = net
 	}
 	publicKeyBytes := pubkey.SerializeCompressed()
-	hash := hash160(publicKeyBytes)[:ripemd160.Size]
+	pkHash := hash160(publicKeyBytes)[:ripemd160.Size]
 	script := make([]byte, 0)
-	script = append([]byte{Op0, byte(len(hash))}, hash...)
-	return &Payment{tmpNet, pubkey, hash, nil, nil, script}
+	script = append([]byte{Op0, byte(len(pkHash))}, pkHash...)
+	witnessHash := sha256.Sum256(script)
+	return &Payment{tmpNet, pubkey, pkHash, nil, nil, script, witnessHash[:]}
 }
 
 // FromPayment creates a Payment struct from a another Payment
 func FromPayment(payment *Payment) (*Payment, error) {
 	if payment.Script == nil {
 		return nil, errors.New("script can't be nil")
 	}
-	redeem := &Payment{payment.Network, payment.PublicKey, payment.Hash, payment.BlindingKey, payment.Redeem, payment.Script}
-	return &Payment{payment.Network, nil, hash160(redeem.Script), nil, redeem, nil}, nil
+	redeem := &Payment{payment.Network, payment.PublicKey, payment.Hash, payment.BlindingKey, payment.Redeem, payment.Script, payment.WitnessHash}
+	return &Payment{payment.Network, nil, hash160(redeem.Script), nil, redeem, nil, nil}, nil
```

Comment:
The `WitnessHash` should not be nil but the sha256 hash of `redeem.Script`:
```suggestion
	return &Payment{payment.Network, nil, hash160(redeem.Script), nil, redeem, nil, sha256.Sum256(redeem.Script)}, nil
```